documentation/modules/ROOT/pages/configuration/applying-transformations-selectively.adoc

```diff
@@ -0,0 +1,143 @@
+You can use the `negate` option to pair the predicate with other transforms that are based on negating the condition.
+
+.Predicate elements
+Predicates include the following elements:
+
+* `predicates` prefix
+* Alias (for example, `isOutboxTable`)
+* Type (for example, `org.apache.kafka.connect.transforms.predicates.TopicNameMatches`).
+  Kafka Connect provides a set of default predicate types, which you can supplement by defining your own custom predicates.
+* Condition statement and any additional configuration properties, depending on the type of predicate (for example, a regex naming pattern)
+
+.Default predicate types
+The following predicate types are available by default:
+
+HasHeaderKey:: Specifies a key name in the header in the event message that you want Kafka Connect to evaluate.
+The predicate evaluates to true for any records that include a header key that has the specified name.
+
+RecordIsTombstone:: Matches records in which the key value of the record is `null`, representing a deleted row in the database.
+Tombstones are fields in a database that indicate that a record is not physically deleted, but has a logical deletion flag that is set to `true` when when the record is marked for deletion.
+The predicate evaluates to `true` for any record that has a `null` key value.
+Use this predicate in combination with a filter SMT to remove tombstone records.
+This predicate has no configuration parameters.
+
+TopicNameMatches:: A regular expression that specifies the name of a topic that you want Kafka Connect to match.
+The predicate is true for connector records in which the topic name matches the specified regular expression.
+Use to apply an SMT according to the name of the topic.
+Topic name of the connect record matches the specified expression.
+
+.Additional resources
+
+* link:https://cwiki.apache.org/confluence/display/KAFKA/KIP-585%3A+Filter+and+Conditional+SMTs[KIP-585: Filter and Conditional SMTs]
+* link:{link-kafka-docs}/#connect_predicates[Apache Kafka documentation for Kafka Connect predicates]
+
+// Type: procedure
+[id="defining-smt-predicates"]
+== Defining SMT predicates
+
+By default, Kafka Connect applies each single message transformation in the {prodname} connector configuration to every change event record that it receives from {prodname}.
+Beginning with Apache Kafka 2.6, you can define an SMT predicate for a transformation in the connector configuration that controls how Kafka Connect applies the transformation.
+The predicate statement defines the conditions under which Kafka Connect applies the transformation to event records emitted by {prodname}.
+Kafka Connect evaluates the predicate statement and then applies the SMT selectively to the subset of records that match the condition that is defined in the predicate.
```

Comment:
Seems a bit redundant with what has been said above?

Reply:
Yeah, true. In the modularized downstream doc, it's possible for readers to land on a page out of context, so for this procedure I intentionally built in a few sentences to orient someone. It's a bit redundant in the upstream context, so maybe best if I conditionalize it as [product] only.